dattri/algorithm/influence_function.py

```diff
@@ -620,35 +649,94 @@ def transform_test_rep(
             torch.Tensor: Transformed test representations. Typically a 2-d
                 tensor with shape (batch_size, transformed_dimension).
         """
-        from dattri.func.fisher import ifvp_datainf
 
-        model_params, param_layer_map = self.task.get_param(ckpt_idx, layer_split=True)
+        def _transform_single_test_rep(
+            v: torch.Tensor,
+            grad: torch.Tensor,
+            regularization: float,
+        ) -> torch.Tensor:
+            """Transformation of a single test representation.
+
+            For any test representation v and training gradient grad, along
+            with regularization term r, DataInf gives:
+            q = (v - (dot(v,grad) / (r + torch.norm(grad) ** 2)) * grad) / r
+
+            Transformed test representations are later used for influence calculation:
+            Inf = dot(g,q) where g is a training representation and q is a transformed
+            test representation.
+
+            Args:
+                v (torch.Tensor): A tensor representing (batched) test set
+                    representation, of shape (num_test,parameter_size).
+                grad (torch.Tensor): A tensor representing a single training
+                    gradient, of shape (parameter_size,).
+                regularization (float): A float default to 0.1. Specifies
+                    the regularization term to be added to the empirical Fisher
+                    information matrix in each layer.
+
+            Returns:
+                torch.Tensor: A tensor corresponding to transformed test representation.
+            """
+            grad = grad.unsqueeze(-1)
+            coef = (v @ grad) / (regularization + torch.norm(grad) ** 2)
+            return (v - coef @ grad.T) / regularization
+
+        regularization = self.transformation_kwargs["regularization"]
```

Comment:
let's directly store the value in self.regularization at initialization and get rid of self. transformation_kwargs, since that's the only elements in this transformation_kwargs

Reply:
Removed kwargs and added new attribute regularization.